### ISSUE TITLE
XSpec occupies xs: and xsl: namespace prefixes

### ISSUE THREAD

**Issue by galtm** [xspec/xspec#639]
The issue is analogous to #466, but for the xs and xsl prefixes.

Reproduction steps:

1. Run `xspec/test/xspec-prefix-conflict.xspec` to test XSLT and XQuery. It passes.
2. Copy `xmlns:x="x-urn:test:xspec-prefix-conflict"` from line 4, paste it, and modify it to read `xmlns:xs="x-urn:test:xspec-prefix-conflict"`.
3. On line 8, change `x:param-mirror-function` to `xs:param-mirror-function`.
4. Run the test again. It errors out.
5. Repeat steps 2-4 with the prefix `xsl` instead of `xs`. It errors out when testing XSLT.

Note: I don't know of a specific use case for using the "xsl" or "xs" prefix for namespaces other than "http://www.w3.org/1999/XSL/Transform" and "http://www.w3.org/2001/XMLSchema", respectively. I only noticed this issue while working on #598 .

**Comment by AirQuick** [comment]
I haven't tested it, but these prefixes might be the case too
https://github.com/xspec/xspec/blob/a5ab424e3b8629efb95b3761029bb0181ba12dd1/src/compiler/generate-xspec-tests.xsl#L48-L52

**Comment by galtm** [comment]
> these prefixes might be the case too

You're probably right. While looking at #777, I found that in `test/xspec-777.xspec`, if I changed the parameter declaration to

```
<x:param as="document-node(element(test))" href="xspec-777.xml" name="impl:test" xmlns:impl="my-namespace"/>
```
then there was an error in the generated XSLT.

**Comment by AirQuick** [comment]
Looks like we'll eventually need to replace `impl:` and `test:` lexical QNames with EQName.

**Comment by galtm** [comment]
Sounds good (whenever someone has the spare time). Maybe I can work on this bug fix, but it wouldn't be that soon.

**Comment by AirQuick** [comment]
Fixed in #982

**Comment by AirQuick** [comment]
Reopening for `xsl`.

**Comment by AirQuick** [comment]
`xsl` fixed in #1099.
